cranelift/codegen/src/isa/x64/inst/mod.rs

```diff
@@ -2191,7 +2191,7 @@ fn x64_get_operands<F: Fn(VReg) -> VReg>(inst: &Inst, collector: &mut OperandCol
         } => {
             collector.reg_use(*idx);
             collector.reg_early_def(*tmp1);
-            collector.reg_early_def(*tmp2);
+            collector.reg_def(*tmp2);
```

Comment:
Can we add a comment here that this is safe because `tmp2` is written only after `idx` is used, and likewise a comment in `emit.rs` in the listing of the emitted sequence that if this property changes, we need to update regalloc metadata?